### PR TITLE
Complete league stats

### PR DESCRIPTION
**Pull request by rrabinovitch**
completed `highest_scoring_visitor`, `highest_scoring_home_team`, `lowest_scoring_visitor`, and `lowest_scoring_home_team`

will refactor later this weekend to eliminate enumerable repetetiveness and create helper methods

@sciencefixion @gabichuelas @evilaspaas1

### CODE REVIEW

**Review by sciencefixion — APPROVED** [review]
Great work! Your commit messages are helpful and I appreciated the comments.

**Review by araspaas — APPROVED** [review]
looks good to me.

**Review by gabichuelas — APPROVED** [review]

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -85,37 +85,123 @@ def average_scores_by_team
...
-    highest_avg_score = average_scores_by_team.max_by do |team, avg_score|
+    highest_avg_score = average_scores_by_team.max_by do |_team, avg_score|
       avg_score
     end
     find_team_by_id(highest_avg_score.first).team_name
   end
 
   def worst_offense
-    lowest_avg_score = average_scores_by_team.min_by do |team, avg_score|
+    lowest_avg_score = average_scores_by_team.min_by do |_team, avg_score|
```
Comment (gabichuelas) [inline lib/stat_tracker.rb:95]: @rrabinovitch what does that `_` before team in the block variable do?

Comment (rrabinovitch) [reply lib/stat_tracker.rb:95]: it's an indicator that that block variable isn't actually utilized in the code block

Comment (rrabinovitch) [reply lib/stat_tracker.rb:95]: so it doesn't actually _do_ anything (as far as I understand); rather it seems to be a conventional thing

**Pull request merged**